### ISSUE TITLE
SC2016 Error Message is Misleading for Multi-line Strings

### ISSUE THREAD

**Issue by jwmrt** [koalaman/shellcheck#2464]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2016
- My shellcheck version (`shellcheck --version` or "online"): 0.8.0
- [x ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash

foo=abc
bar=xyz
declare -r str='
    foo: "'$foo'",
    blah: "blah",
    blah: "blah",
    blah: "blah",
    "bar:" "$bar",
}'

echo "$str"
echo $bar


```

#### Here's what shellcheck currently says:

In /tmp/x.sh line 6:
    foo: "'$foo'",
               ^-- SC2016 (info): Expressions don't expand in single quotes, use double quotes for that.

#### Here's what I wanted or expected to see:

In /tmp/x.sh line 10:
    foo: "$bar",
              ^-- SC2016 (info): Expressions don't expand in single quotes, use double quotes for that.

#### Rationale:
The error message is confusing because it looks like shellcheck is warning about $foo which is NOT within single quotes, whereas the real error is with bar on line 10. This also affects backtick evaluation and I suspect there are several open issues about SC2016 "false positives" that actually are errors but are being misread because of the line number (and column) that shellcheck is reporting. In particular, I think this issue ( https://github.com/koalaman/shellcheck/issues/1693 ) is an example of this (i.e. the problem there is actually the backticks later in the string--of course, in that example the user probably wants to ignore that anyways, but it's not a false positive related to variable expansion as the error message seems to indicate)
